File: bitcoin/src/main/java/com/brentpanther/bitcoinwidget/CoinSelectionView.java
package com.brentpanther.bitcoinwidget;

import android.content.Context;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;


public class CoinSelectionView extends LinearLayout {

    private Coin coin;
    private CoinSelectedListener listener;

    public CoinSelectionView(Context context) {
        super(context);
        inflate(context, R.layout.view_coin_selector, this);
        int height = (int)getResources().getDimension(R.dimen.coin_selection_height);
        MarginLayoutParams layoutParams = new MarginLayoutParams(LayoutParams.MATCH_PARENT, height);
        int verticalMargin = (int) getResources().getDimension(R.dimen.vertical_margin);
        int horizontalMargin = (int) getResources().getDimension(R.dimen.horizontal_margin);
        layoutParams.setMargins(horizontalMargin, verticalMargin, horizontalMargin, verticalMargin);
        setLayoutParams(layoutParams);
        setOnClickListener(v -> listener.selected(coin));
        setBackgroundResource(R.drawable.bg_rounded);
    }

    public void setCoin(Coin coin, CoinSelectedListener listener) {
        this.coin = coin;
        this.listener = listener;
        TextView name = findViewById(R.id.coin_name);
        name.setText(coin.getName());
        ImageView image = findViewById(R.id.coin_icon);
        image.setImageResource(coin.getIcon());
    }

    interface CoinSelectedListener {
        void selected(Coin coin);
    }
}

File: bitcoin/src/main/java/com/brentpanther/bitcoinwidget/CoinSelectionView.kt
package com.brentpanther.bitcoinwidget

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView


class CoinSelectionView(context: Context) : LinearLayout(context) {

    private lateinit var coin: Coin
    private var listener: ((coin: Coin) -> kotlin.Unit)? = null

    init {
        View.inflate(context, R.layout.view_coin_selector, this)
        val height = resources.getDimension(R.dimen.coin_selection_height).toInt()
        val layoutParams = ViewGroup.MarginLayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, height)
        val verticalMargin = resources.getDimension(R.dimen.vertical_margin).toInt()
        val horizontalMargin = resources.getDimension(R.dimen.horizontal_margin).toInt()
        layoutParams.setMargins(horizontalMargin, verticalMargin, horizontalMargin, verticalMargin)
        setLayoutParams(layoutParams)
        setOnClickListener { listener?.invoke(coin) }
        setBackgroundResource(R.drawable.bg_rounded)
    }

    fun setCoin(coin: Coin, listener: ((coin: Coin) -> kotlin.Unit)?) {
        this.coin = coin
        this.listener = listener
        val name = findViewById<TextView>(R.id.coin_name)
        name.text = coin.coinName
        val image = findViewById<ImageView>(R.id.coin_icon)
        image.setImageResource(coin.icon)
    }

}
